### PR TITLE
Issue #520: Made https://devguide.python.org/committing/ simpler

### PR DESCRIPTION
**Pull request by jablonskidev**
These changes are based on issue #520.

Issue #520 was somewhat open ended, so I generally tried to make the text easier to skim so that core developers could use it as a quick reference. 

I'm still learning reStructuredText, so there are some issues with formatting, some of which have caused problems with line lengths in vertical lists.

I'd be interested to know if this kind of simplification is what you were going for. I'm looking forward to hearing how we can keep improving this resource.

### CODE REVIEW

File: committing.rst
Comment (methane) [inline committing.rst:42]: Should core developers run patchcheck? Doesn't CI run patchcheck?

File: committing.rst
Comment (methane) [inline committing.rst:16]:
```suggestion

* **Are there ongoing discussions in b.p.o.?**
   Read the linked b.p.o. issue and check the pull request is ready to merge. If there are ongoing discussions, we need to make a consensus before merging the pull request.
```

File: committing.rst
Comment (jablonskidev) [reply committing.rst:42]: @methane Thanks for your feedback. I included this item in the checklist because the current version of this page includes the following section:

```
Patch checklist
'''''''''''''''

You should also :ref:`run patchcheck <patchcheck>` to perform a quick
sanity check on the changes.
```

If core developers don't need to run `patchcheck`, then I have no objection to removing the checklist item.

File: committing.rst
Comment (jablonskidev) [reply committing.rst:16]: @methane I added a modified version of this in a commit. Let me know what you think.

**Comment by jablonskidev** [comment]
@brettcannon I'd be interested to know if the changes that I'm proposing with this PR are what you were going for when you said that the [page on accepting pull requests](https://devguide.python.org/committing/) needed to be simplified.

File: committing.rst
Comment (jablonskidev) [reply committing.rst]: @methane I removed this item from the checklist.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
Please revert the removal of the patchcheck advice and edit it instead.

Patchcheck is mainly checking for extra whitespace at the end of lines and files.  The devguidev should say that.  If CI patchcheck fails, CI fails.  So people *should* run patchcheck or the equivalent before making a PR, just as they should run unittests before doing so, and not leave it to CI to catch test failures even on their development OS.

File: committing.rst
Comment (terryjreedy) [reply committing.rst]: I consider this a mistake.  See review.

**Comment by jablonskidev** [comment]
@terryjreedy Thanks for your feedback. I put that checklist item back in and modified it slightly.

**Comment by methane** [comment]
> Patchcheck is mainly checking for extra whitespace at the end of lines and files. The devguidev should say that. If CI patchcheck fails, CI fails. So people _should_ run patchcheck or the equivalent before making a PR, just as they should run unittests before doing so, and not leave it to CI to catch test failures even on their development OS.

I agree that devguide should note about run patchcheck.

But I disagree about adding it in "Assessing a pull request" section. Core developers don't have to run patchcheck manually before merging the PR if CI status is green, do it?

**Comment by terryjreedy** [comment]
No.  Running patchcheck locally, like running unittests locally, is something to do before making a PR, so that the future PR is likely to pass CI tests at least on the same OS the patch was developed on.  It is frustrating when an otherwise good PR is blocked waiting for a trailing whitespace fix.

**Comment by methane** [comment]
> No. Running patchcheck locally, like running unittests locally, is something to do before making a PR, so that the future PR is likely to pass CI tests at least on the same OS the patch was developed on. It is frustrating when an otherwise good PR is blocked waiting for a trailing whitespace fix.

"Assessing a pull request" is the section for "reviewing" PRs. Not for "creating" PRs.

What should be done when making PR is documented in https://devguide.python.org/pullrequest/ already.

**Review by brettcannon — CHANGES_REQUESTED** [review]
Looks good overall, but there's some simplifications that I think we can make.

File: committing.rst
Comment (jablonskidev) [reply committing.rst]: @brettcannon Thanks for your feedback! I tweaked the formatting and wording in your suggestion and then committed my own version. Please let me know what you think.

File: committing.rst
Comment (jablonskidev) [reply committing.rst]: @brettcannon There was already a checklist item for the test suite, so I combined the two checklist items and removed the specific mention of `patchcheck`.

File: committing.rst
Comment (jablonskidev) [reply committing.rst]: @brettcannon I removed this item. Happy to hear that you're listening to the _Real Python_ podcast 😄

File: committing.rst
Comment (jablonskidev) [reply committing.rst]: @brettcannon I removed this item too. Thanks for all your feedback. I'm looking forward to getting this out! 

If there are any other parts of the documentation that you specifically wish someone was working on, then please feel free to let me know. I'm looking forward to the documentation work group getting started, so I'm picking away at issues in the meantime.

File: committing.rst
Comment (brettcannon) [reply committing.rst]: I actually didn't get it from the podcast, just from reviewing your zipimport PR. :)

**Comment by jablonskidev** [comment]
@brettcannon Are there any other changes that you'd like me to make?

**Comment by jablonskidev** [comment]
@Mariatta I made the changes that you requested in #655. Could you please take a look and let me know if you'd like me to make improvements?

**Comment by brettcannon** [comment]
@jablonskidev only if it involves spacetime to give me more so I can find the team to do another review. 😉 This is at the top of my PR review list.

File: committing.rst
Comment (brettcannon) [inline committing.rst:168]: Having to update this doc every year just to change the maintenance branch numbers seems like wasted work when the list of maintenance branches is on the home page for the devguide itself at https://devguide.python.org/#status-of-python-branches.

```suggestion
* **You should not commit directly into the** ``master`` **branch, or any of the maintenance branches.**
```

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
Thanks, @jablonskidev !

**Pull request merged**
